nix/module.nix

```diff
@@ -30,47 +27,32 @@ in {
         '';
       };
 
-    package = mkOption {
-      type = types.path;
-      default = defaultHyprlandPackage;
-      defaultText = literalExpression ''
-        hyprland.packages.''${pkgs.stdenv.hostPlatform.system}.default.override {
-          enableXWayland = config.programs.hyprland.xwayland.enable;
-          hidpiXWayland = config.programs.hyprland.xwayland.hidpi;
-          inherit (config.programs.hyprland) nvidiaPatches;
-        }
-      '';
-      example = literalExpression "pkgs.hyprland";
+    package = mkPackageOptionMD inputs.packages.${pkgs.system} "hyprland" {};
+
+    finalPackage = mkOption {
+      type = types.package;
+      readOnly = true;
+      default = cfg.package.override {
+        enableXWayland = cfg.xwayland.enable;
+        enableNvidiaPatches = cfg.enableNvidiaPatches;
+      };
+      defaultText =
+        literalExpression
+        "`programs.hyprland.package` with applied configuration";
       description = mdDoc ''
-        The Hyprland package to use.
-        Setting this option will make {option}`programs.hyprland.xwayland` and
-        {option}`programs.hyprland.nvidiaPatches` not work.
+        The Hyprland package after applying configuration.
       '';
     };
 
-    xwayland = {
-      enable = mkEnableOption (mdDoc "XWayland") // {default = true;};
-      hidpi =
-        mkEnableOption null
-        // {
-          description = mdDoc ''
-            Enable HiDPI XWayland, based on [XWayland MR 733](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/733).
-            See <https://wiki.hyprland.org/Nix/Options-Overrides/#xwayland-hidpi> for more info.
-          '';
-        };
-    };
+    portalPackage = mkPackageOptionMD inputs.xdph.packages.${pkgs.system} "xdg-desktop-portal-hyprland" {};
```

Comment:
If XDPH may receive its own options in the future, consider `portal.package`. This may seem to conflict with my earlier suggestion pertaining to `xwayland.enable`, but `xwayland` no longer has any other options.

Reply:
We'll be adding that if it is the case. Currently, no such options are planned. Also I'm not aware of any portal that can be configured in any way.

Reply:
In `hyprland.portal`, I do override `UseIn`: <https://github.com/hyprwm/xdg-desktop-portal-hyprland/blob/master/hyprland.portal#L4>

That could be exposed as an option. I do this: <https://github.com/spikespaz/dotfiles/blob/master/hm-modules/desktop-portals.nix>

There was recently a PR to `xdg-desktop-portal-wlr` that removed the necessity for this, something to do with precedence. I can't seem to find it though, but maybe I'm looking at the wrong thing.

This issue seems relevant: <https://github.com/emersion/xdg-desktop-portal-wlr/issues/268>

Reply:
Oh, I see. From past issues, it's more likely that the GNOME portal causes issues and would need that, and not Hyprland. I think it's best to wait until the new mechanism arrives.